### PR TITLE
Podspawn admin verb, like Spawn, but more IC

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
admin: New 'Podspawn' verb, which functions like 'Spawn', except
any atoms movable spawned will be dropped in via a no-damage, no-explosion
Centcom supply pod.
/:cl:

Essentially, sometimes you want to spawn stuff in, quickly, in an adhoc manner,
while trying to keep some level of immersion. Random items appearing can be
a little silly at times.

Use of the full Centcom launchbay is fine if you're doing a full blown drop or event
or want some customisation, but sometimes you want a quick supply pod.

The admin smite "Supply Pod (Quick)" has been used for this purpose, but it has a side
effect of setting people on fire and damaging them, which isn't what you want
if it's just for spawning stuff in.

### CODE REVIEW

File: code/modules/cargo/supplypod.dm [hunk 1]
```diff
@@ -20,7 +20,7 @@
 	var/bluespace = FALSE //If true, the pod deletes (in a shower of sparks) after landing
 	var/landingDelay = 30 //How long the pod takes to land after launching
 	var/openingDelay = 30 //How long the pod takes to open after landing
-	var/departureDelay = 30 //How long the pod takes to leave after opening (if bluespace=true, it deletes. if reversing=true, it flies back to centcom)
+	var/departureDelay = 30 //How long the pod takes to leave after opening (if bluespace=true, it deletes. if reversing=true, it flies back to centcom
```
Comment (MrDoomBringer) [inline code/modules/cargo/supplypod.dm]:
```suggestion
	var/departureDelay = 30 //How long the pod takes to leave after opening. If bluespace = TRUE, it deletes. If reversing=TRUE, it flies back to centcom)
```

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

**Comment by MrDoomBringer** [comment]
Could we make a link to this proc in the game panel? So next to the game panel's "Spawn" button, there would be "Pod Spawn"?

**Comment by coiax** [comment]
Yes. It's already in the game panel, under Pod Spawn.

**Comment by MrDoomBringer** [comment]
@coiax sorry i was unclear, i was thinking something more like this 
image directly in the create object and create mob windows

**Comment by coiax** [comment]
Oh sure.

**Comment by elyina** [comment]
Very cool, I've been wanting something like this for years. I always prefer to try to spawn things in in an IC way.

**Comment by MrDoomBringer** [comment]
@elyina i mean you could also use the centcom podlauncher but yeah this is much faster

**Comment by coiax** [comment]
@MrDoomBringer 
image

Will this do?

**Comment by MrDoomBringer** [comment]
@coiax That looks good!

**Comment by michnia** [comment]
Dnia 21 grudnia 2018 13:47 MrDoomBringer   napisał(a):  github.com @coiax  That looks good!

**Comment by coiax** [comment]
@ExcessiveUseOfCobblestone

**Comment by ExcessiveUseOfCobblestone** [comment]
redditor

File: code/modules/cargo/supplypod.dm [hunk 2]
```diff
@@ -200,7 +199,7 @@
 /obj/structure/closet/supplypod/Destroy()
 	if (!opened) //If we havent opened yet, we're opening because we've been destroyed. Lets dump our contents by opening up
 		open(src, broken = TRUE)
-	return ..()
+	. = ..()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/cargo/supplypod.dm]: lol why did you change this

Comment (coiax) [reply code/modules/cargo/supplypod.dm]: Force of habit, sorry. At some point Remie told me that . was faster than using return.

**Comment by Jordie0608** [comment]
Conflict.

File: code/modules/cargo/supplypod.dm [hunk 3]
```diff
@@ -79,7 +79,8 @@
 	update_icon()
 
 /obj/structure/closet/supplypod/tool_interact(obj/item/W, mob/user)
-	if (bluespace) //We dont want to worry about interacting with bluespace pods, as they are due to delete themselves soon anyways.
+	if(bluespace) //We dont want to worry about interacting with bluespace pods, as they are due to delete themselves soon anyways.
```
Comment (MrDoomBringer) [inline code/modules/cargo/supplypod.dm]: hmm yes quite

File: code/modules/cargo/supplypod.dm [hunk 4]
```diff
@@ -79,7 +79,8 @@
 	update_icon()
 
 /obj/structure/closet/supplypod/tool_interact(obj/item/W, mob/user)
-	if (bluespace) //We dont want to worry about interacting with bluespace pods, as they are due to delete themselves soon anyways.
+	if(bluespace) //We dont want to worry about interacting with bluespace pods, as they are due to delete themselves soon anyways.
+	if(bluespace) //We dont want to worry about interacting with bluespace pods, as they are due to delete themselves soon anyways.
```
Comment (MrDoomBringer) [inline code/modules/cargo/supplypod.dm]: hmm yes quite

**Review by MrDoomBringer — APPROVED** [review]

File: code/modules/admin/admin.dm [hunk 5]
```diff
@@ -651,6 +651,29 @@
...
+	set name = "Podspawn"
+
+	if(!check_rights(R_SPAWN))
+		return
+
+	var/chosen = pick_closest_path(object)
+	if(!chosen)
+		return
+	if(ispath(chosen, /turf))
+		var/turf/T = get_turf(usr.loc)
```
Comment (Cyberboss) [inline code/modules/admin/admin.dm]:
```suggestion
 		var/turf/T = get_turf(usr)
```

File: code/modules/cargo/supplypod.dm (hunk 4 above)
Comment (Cyberboss) [inline code/modules/cargo/supplypod.dm]: fix

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by coiax** [comment]
Is this good to go?

**Review by Cyberboss — APPROVED** [review]
